Remove Quirk shouldAvoidUsingIOS13ForGmail for mail.google.com

karlcow opened the pull request:
#### eab1695be7212b006c689aa7ef18988beb685925
<pre>
Remove Quirk shouldAvoidUsingIOS13ForGmail for mail.google.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=260708">https://bugs.webkit.org/show_bug.cgi?id=260708</a>
<a href="https://rdar.apple.com/114438822">rdar://114438822</a>

Reviewed by Brent Fulgham.

This is now dead code. The current release of IOS is 18,
and this was addressing an issue for iOS 13.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAvoidUsingIOS13ForGmail const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::platformUserAgent const):

Canonical link: <a href="https://commits.webkit.org/282652@main">https://commits.webkit.org/282652@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7f7c34c678511d0817793d763d6dd11e1228a0ad

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63787 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43144 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16384 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67808 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14395 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65907 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50834 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14675 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51401 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9954 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66856 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39972 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55219 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32018 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36652 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12598 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13268 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/58544 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12925 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69504 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7734 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12482 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58727 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7767 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55313 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58873 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14118 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6438 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38964 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40043 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41226 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39786 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow commented on Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
```diff
@@ -4941,10 +4938,6 @@ static bool selectionIsInsideFixedPositionContainer(LocalFrame& frame)
     if (!document)
         return String();
 
-    if (osNameForUserAgent() == "iPhone OS"_s) {
-        if (document->quirks().shouldAvoidUsingIOS13ForGmail())
-            return standardUserAgentWithApplicationName({ }, "12_1_3"_s);
-    }
```
@brentfulgham For the reviewer, the initial patch was adding more code than what I removed here. Not sure if I should revert everything as it was initially. 
https://github.com/WebKit/WebKit/commit/b49d99266d0d84c0331d69f63263f5a856a5e773

brentfulgham replied:
I don't see any cases where this method returns anything *but* `String()`. I think you should simplify to match `WebPageMac.mm` and just return `String()`. Extra credit: Move the method in `WebPageMac.mm` to `WebPageCocoa.mm` and delete this entirely.

karlcow replied:
This is declared in `WebPage.h`
https://github.com/WebKit/WebKit/blob/c2e4b673742acda4d321dbfa88437ae52fcda2ab/Source/WebKit/WebProcess/WebPage/WebPage.h#L584

Then used in `WebPagePlayStation.cpp`

https://github.com/WebKit/WebKit/blob/c2e4b673742acda4d321dbfa88437ae52fcda2ab/Source/WebKit/WebProcess/WebPage/playstation/WebPagePlayStation.cpp#L56-L62

and `WebPageWin.cpp`

https://github.com/WebKit/WebKit/blob/c2e4b673742acda4d321dbfa88437ae52fcda2ab/Source/WebKit/WebProcess/WebPage/win/WebPageWin.cpp#L77-L80

and `WebPageGLib.cpp`

https://github.com/WebKit/WebKit/blob/c2e4b673742acda4d321dbfa88437ae52fcda2ab/Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp#L201-L207

and `WebPageIOS.mm`

https://github.com/WebKit/WebKit/blob/c2e4b673742acda4d321dbfa88437ae52fcda2ab/Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm#L4929-L4949

So here if I just return `String()` like others that seems to work without issues. 
There's a strange quirk which is described in rdar://116201535



and `WebPageMac.mm`

https://github.com/WebKit/WebKit/blob/c2e4b673742acda4d321dbfa88437ae52fcda2ab/Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm#L799-L802

karlcow replied:
the FIXME was added by @achristensen07 
in https://github.com/WebKit/WebKit/commit/57435dd4c7a3967e2fa9eb4da36183dc061d7ea5

and the code was added by @achristensen07 
https://github.com/WebKit/WebKit/commit/93cc4767ec04fbe1027cda51ccb2ad1d47fa879f

in the context of site isolation
https://github.com/WebKit/WebKit/commit/93cc4767ec04fbe1027cda51ccb2ad1d47fa879f#diff-49068a67a1cd1b7b267adbee4e47477a00787e24f9cfcbd6df2372be36b5778aL4467-R4479

but I do not see a rationale for it.

brentfulgham replied:
@achristensen07 reached out privately and said that you should ignore that FIXME.

brentfulgham replied:
@karlcow I think you should finish this patch using the simple `return String()` (or `return { }`) so that iOS and macOS match. In a follow-up patch, I think we might want to move to the form Playstation and Glib use so that we can fully move to the UA Quirks living in WebKit (instead of at the browser layer). That would allow us to collapse all platforms to a single implementation at the `WebPage` level, with the quirks living in the `WebCore::standardUserAgentForURL(...)` method.

karlcow replied:
Good idea! Thanks. This is resolved.

karlcow committed:
Remove Quirk shouldAvoidUsingIOS13ForGmail for mail.google.com
https://bugs.webkit.org/show_bug.cgi?id=260708
rdar://114438822

Reviewed by Brent Fulgham.

This is now dead code. The current release of IOS is 18,
and this was addressing an issue for iOS 13.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAvoidUsingIOS13ForGmail const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::platformUserAgent const):

Canonical link: https://commits.webkit.org/282652@main
